Add options to toggle views between deck/list view on PageSingleSet

DangMin opened the pull request:
### Tasks:
- [x] Add buttons to toggle between view options (using `ButtonGroup`
- [x] Add `ButtonGroup` component
- [x] Provide UI for the views

DangMin committed:
Add UI to toggle between deck view and list view for PageSingleSet

- Change addCardToSet parameter: add userId as database changed locally
- Export more Icons
- Add function to store view options to localStorage
- Add ButtonGroup component

DangMin committed:
More UI addition and adjustment.

DangMin commented:
Sneak peak of current UI version:
![Screenshot_20220502_215307](https://user-images.githubusercontent.com/7398845/166307606-76698c7f-089f-4a58-843b-e43aa5f2e1b4.png)

DangMin committed:
Minor UI changes.

DangMin committed:
Add Tooltip component.

DangMin commented:
Most recent commit: "Add Tooltip component"

file:///home/dangmin/Pictures/Screenshot_20220507_192747.png

DangMin committed:
Add functions to useSingleSet.

kcjpop commented on web/lib/components/common/ButtonGroup.jsx:
```diff
@@ -0,0 +1,20 @@
+        cloneElement(child, {
+          key: `button-${index}`,
+          className: `border border-slate-300 box-border ${
+            index === 0
+              ? 'rounded-md !rounded-r-none'
+              : index === children.length - 1
+              ? 'rounded-md !rounded-l-none'
+              : '!rounded-none'
```
Let's `import clsx from 'clsx'` and use it instead of nested ternary operator?

kcjpop commented on web/lib/components/common/Tooltip.jsx:
```diff
@@ -0,0 +1,70 @@
+            ref={arrowRef}
+            className="absolute h-3 w-6 bg-slate-700"
+            style={{
+              top: arrowY ?? '',
+              left: arrowX ?? '',
```
I'm not a fan of arrow tooltip tho 😄

kcjpop commented on web/lib/components/flashcards/FlashcardRow.jsx:
```diff
@@ -0,0 +1,36 @@
+import { VolumnIcon, TrashIcon, EditIcon } from '../common/Icons'
+
+export function FlashcardRow({ flashcard, editable }) {
+  console.log({ flashcard })
```
Should remove.

kcjpop commented on web/lib/components/flashcards/FlashcardRow.jsx:
```diff
@@ -0,0 +1,36 @@
+      key={flashcard.id}
+      className="grid grid-cols-3 grid-rows-1 rounded-lg bg-slate-100 p-2 shadow hover:bg-slate-200"
+      style={{
+        gridTemplateColumns: '1fr 1fr min-content',
```
Then we don't need `grid-cols-3` above.

kcjpop reviewed (approved):
Looking good so far. I think `DECK_MODE` should be the default one.

DangMin committed:
Resolve PR conversation

- DECK_MODE as default view
- Remove arrow from Tooltip
- Add rule 'no-console', only allow "console.error"
- Remove console.log
